modules/services/password-store-sync.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.services.password-store-sync = {
+    enable = mkEnableOption "Password store periodic sync";
+
+    frequency = mkOption {
+      type = types.str;
+      default = "*:0/5";
+      description = ''
+        How often to synchronise the password store git repository.
+        This value is passed to the systemd timer configuration as the onCalendar option.
+        See
+        <citerefentry>
+          <refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum>
+        </citerefentry>
+        for more information about the format.
+      '';
+    };
+  };
+
+  config = let
+    serviceCfg = config.services.password-store-sync;
+    programCfg = config.programs.password-store;
+
+    # from
+    # https://github.com/NixOS/nixpkgs/blob/release-19.03/nixos/modules/system/boot/systemd-lib.nix
+    # https://github.com/NixOS/nixpkgs/blob/release-19.03/nixos/modules/system/boot/systemd.nix
+    makeJobScript = let
+      shellEscape = s: (lib.replaceChars [ "\\" ] [ "\\\\" ] s);
+      mkScriptName = s: "unit-script-" + (lib.replaceChars [ "\\" "@" ] [ "-" "_" ] (shellEscape s) );
+      mkScriptContent = script: ''
+        #! ${pkgs.runtimeShell} -e
+        ${script}
+      '';
+    in name: text: toString (pkgs.writeTextFile {
+      name = mkScriptName name;
+      executable = true;
+      text = mkScriptContent text;
+    });
+
+    makeEnvironmentPairs = lib.mapAttrsToList (key: value: builtins.toJSON "${key}=${value}");
+  in mkIf serviceCfg.enable {
+    systemd.user.services.password-store-sync = {
+      Unit = {
+        Description = "Password store sync";
+        PartOf = [ "network-online.target" ];
+      };
+      Service = {
+        CPUSchedulingPolicy = "idle";
+        IOSchedulingClass = "idle";
+        Environment = makeEnvironmentPairs programCfg.settings;
+        ExecStart = makeJobScript "password-store-sync" ''
+          ${pkgs.pass}/bin/pass git pull --rebase && \
```

Comment:
Could you add a configurable remote?

Reply:
Is it possible to specify an SSH Key? Like with the "core.sshCommand" config?

Reply:
I'm not sure whether it's useful to expose those parameters through module options as they can already be specified by setting an upstream and a `git config` entry respectively.

Reply:
I see, that's simpler indeed! Shouldn't we add a comment in the documentation about that then? :wink:

Reply:
I'd like to add that to the description of the `enable` option but `mkEnableOption` makes it annoying.

Reply:
Worst comes to worst, we can still add it to the `config` option, since its name implies generic configuration of password-store

Reply:
The `settings` option belongs to the program module, far away from the synchronisation service module.

Reply:
Oh yeah, that was dumb of me :-/

Reply:
Clarified by rephrasing the description of the `frequency` option.